go-controller/pkg/node/gateway.go

```diff
@@ -226,6 +246,17 @@ func gatewayInitInternal(nodeName, gwIntf, egressGatewayIntf string, gwNextHops
 		}
 	}
 
+	if config.Default.EnableUDPAggregation {
+		err = setupUDPAggregationUplink(gatewayBridge.uplinkName)
```

Comment:
does this need to be set on the actual physical nic too? In this case its setting it on the bridge interface, but no the physical nic?

Reply:
`gatewayBridge.uplinkName` is the physical nic; `gatewayBridge.bridgeName` is the bridge.

(it needs to be set on the physical NIC but not on the bridge)

Reply:
yes, we want rx-udp-gro-forwarding on the physical NIC, on the host veth device and nowhere else.

When traversing:

veth -> host veth-> ovs ->mp0 ->br-ex ->eth0
        
GRO will happen on the host veth, and the aggregate packet will traverse all the remaining stack.

In the opposite direction GRO will take place on the phy NIC and aggregate packets will traverse the full stack up to the destination socket. They will be segmented again just before being enqueued into such socket.

Reply:
@pabeni so the question is do you need it set on intermediary hops/interfaces in the stack? From your answer it sounds like no, but just want to make sure. The traffic will go from the veth into OVS, then into the host via a kernel interface called mp0. From there the host will route the packet out the br-ex interface back into another OVS bridge, and at that point leave the physical nic.

Reply:
You don't need to set it on intermediary hops, because the intermediary hops don't need to do anything special. (To them it just looks like the endpoint had sent a single large UDP packet rather than several smaller ones.)

Reply:
@trozet: as Dan said, no need to apply any configuration an the "intermediate" devices. All the bridge/tunnel/host port will leave the packets unmodified.